Dockerfile

```diff
@@ -1,19 +1,22 @@
 ####################################################################
-# base image contains all the dependencies we need to build the code
-FROM valhalla/valhalla:build-3.1.5 as builder
+FROM ubuntu:20.04 as builder 
 MAINTAINER Kevin Kreiser <kevinkreiser@gmail.com>
 
-# get the code into the right place and prepare to build it
+# set paths
+ENV PATH /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:$PATH
+ENV LD_LIBRARY_PATH /usr/local/lib:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu:/lib32:/usr/lib32
+
+# install deps
 WORKDIR /usr/local/src/valhalla
+COPY ./scripts/install-linux-deps.sh /usr/local/src/valhalla/scripts
+RUN bash ./scripts/install-linux-deps.sh
+
+# get the code into the right place and prepare to build it
 ADD . /usr/local/src/valhalla
 RUN ls
 RUN git submodule sync && git submodule update --init --recursive
 RUN rm -rf build && mkdir build
 
-# upgrade Conan:
-# https://github.com/valhalla/valhalla/issues/3685#issuecomment-1198604174
-RUN pip install --upgrade conan
-
```

Comment:
on the other hand @ImreSamu : if we remove this line, we wouldn't actually prevent issue #3685 for docker builds. as long as the installation script doesn't change, locally cached layers will be used. I think it's best to keep it

Reply:
> locally cached layers will be used.

the ["old" documentation](https://github.com/valhalla/valhalla/blob/3ecf1c7d87854e9317113c1f67504f95a7b4ddc8/docker/README.md#to-buildpublish-images-manually
)  contains the `--no-cache` suggestion, probably we should also add the `--pull`
*  `docker build --pull --no-cache ...`
   *  `--no-cache`  "Do not use cache when building the image"
   *  `--pull` : "Always attempt to pull a newer version of the image"

probably the circle-CI `build-docker-images` job should contain these 2 extra safety parameters.
* now: `docker build --tag valhalla/valhalla:run-${BUILD_TAG} .`
* proposed: `docker build  --pull --no-cache  --tag valhalla/valhalla:run-${BUILD_TAG} .`

Reply:
I'm not so concerned about circleci builds actually, more about user builds bcs we don't cache docker in ci, but other machines do. though we could pull the newest base image, true, I'll add that.